### PR TITLE
Refactor image commands

### PR DESCRIPTION
**Pull request by Daniel-GrunbergerCA**
## Overview
This PR refactors the kubescape commands so that the `scan` command (for an image) and the new `patch` command (introduced [here](https://github.com/kubescape/kubescape/pull/1332)) will be under a new command called `image`:
1. `kubescape image scan <image>`
2. `kubescape image patch <image>`

It also refactors the `image scan` logic so that the main logic for the scan will be on the `core` package, as in other commands.

## How to Test

Please refer to the `README.md` file under the `cmd/image` directory to see how the commands can be tested and how the output looks like.

### CODE REVIEW

File: core/cautils/normalize_image_name.go [hunk 1]
```diff
@@ -1,10 +1,10 @@
-package opaprocessor
+package cautils
 
 import (
 	"github.com/docker/distribution/reference"
 )
 
-func normalize_image_name(img string) (string, error) {
+func NormalizeImageName(img string) (string, error) {
```
Comment (dwertent) [inline core/cautils/normalize_image_name.go:7]: @YiscahLevySilas1 I missed your reply, does this affect the latest rego changes?

Comment (YiscahLevySilas1) [reply core/cautils/normalize_image_name.go:7]: the rego uses [image.parse_normalized_name](https://github.com/kubescape/regolibrary/blob/623bec0346a1caed310a7c3832f5f1640ca0f085/rules/container-image-repository-v1/raw.rego#L8).

afaik, as far as the [declaration](https://github.com/kubescape/kubescape/blob/076aa7f8fe66e183f61b4023ea1f9b85751d0704/core/pkg/opaprocessor/utils.go#L100), is [alligned](https://github.com/kubescape/kubescape/blob/076aa7f8fe66e183f61b4023ea1f9b85751d0704/core/pkg/opaprocessor/processorhandler.go#L294) with the [definition](https://github.com/kubescape/kubescape/blob/076aa7f8fe66e183f61b4023ea1f9b85751d0704/core/pkg/opaprocessor/utils.go#L104) it should work

File: go.mod [hunk 2]
```diff
@@ -422,3 +457,6 @@ replace (
 	// TODO(vladklokun): armoapi-go requires gorm v1.24.6, but Grype depends on v1.23.10 and breaks on anything above. We can’t have both. Upstream changes to Grype?
 	gorm.io/gorm => gorm.io/gorm v1.23.10 // indirect
 )
+
+// TODO(anubhav06): Remove this once we have a release of copacetic with the support for patching kubescape image scan results.
```
Comment (dwertent) [inline go.mod:461]: @Daniel-GrunbergerCA @anubhav06 please remove this replace

Comment (anubhav06) [reply go.mod:461]: The integration between copa and kubescape is not live yet. For the meantime, I have added the integration between copa and kubescape in a fork of mine(of copa).
I'm actively working with the copa team on the feature of incorporating multiple scanners in copa. Once we have a release of copa which supports kubescape, we can remove this.

File: core/cautils/normalize_image_name.go (hunk 1 above)
Comment (Daniel-GrunbergerCA) [reply core/cautils/normalize_image_name.go:7]: yes it will not affect

File: go.mod (hunk 2 above)
Comment (craigbox) [reply go.mod:461]: I've never seen the license check pass in the GitHub Action.  Someone should fix this, but we also have FOSSA, which says everythign is OK.

Comment (anubhav06) [reply go.mod:461]: I've also made the [PR#1374](https://github.com/kubescape/kubescape/pull/1374) which changes reference back to copa, and in that version copa has Apache-2.0-License (previously MIT), incase that helps in fixing the check.

**Comment by matthyx** [comment]
@Daniel-GrunbergerCA needs rebase

**Pull request closed without merging**